[BUG] `M503 C` does not write configuration files to SD card.

### Did you test the latest `bugfix-2.1.x` code?

Yes, and the problem still exists.

### Bug Description

When building a firmware with `CONFIGURATION_EMBEDDING` enabled, the configuration of that firmware is supposed to be written to the SD Card when `M503 C` is used.

Currently, no configuration files are written to the SD card.

```
Recv: Cap:CONFIG_EXPORT:1
...
Send: M503 C
Recv: echo:; Linear Units:
Recv: echo:  G21 ; (mm)
...
Send: M22
Recv: echo:SD card released
```

No files are written to the SD card as expected.

### Bug Timeline

_No response_

### Expected behavior

As per documentation:
https://marlinfw.org/docs/gcode/M503.html

[C] 2.0.9.3 `CONFIGURATION_EMBEDDING` | Save the embedded configuration ZIP file to the SD Card or Flash Drive.

### Actual behavior

No files are written.

### Steps to Reproduce

_No response_

### Version of Marlin Firmware

bugfix-2.1.x

### Printer model

_No response_

### Electronics

_No response_

### Add-ons

_No response_

### Bed Leveling

None

### Your Slicer

None

### Host Software

None

### Don't forget to include

- [X] A ZIP file containing your `Configuration.h` and `Configuration_adv.h`.

### Additional information & file uploads

_No response_

Doesn't work.  Just created a new compile error

Using default config

enable
EEPROM_SETTINGS
SDSUPPORT
CONFIGURATION_EMBEDDING

add
FORCE_CONFIG_EMBED

compiling errors with

```
Marlin/src/gcode/eeprom/M500-M504.cpp: In static member function 'static void GcodeSuite::M503()':
Marlin/src/gcode/eeprom/M500-M504.cpp:72:64: error: no matching function for call to 'SdBaseFile::open(MediaFile, const char [7], int)'
         if (file.open(card.getroot(), "mc.zip", O_WRITE|O_CREAT) && file.write(pgm_read_ptr(mc_zip), size) != -1 && file.close())

```

--- a/Marlin/src/gcode/eeprom/M500-M504.cpp
+++ b/Marlin/src/gcode/eeprom/M500-M504.cpp
@@ -66,11 +66,19 @@ void GcodeSuite::M502() {
 
     #if ENABLED(CONFIGURATION_EMBEDDING)
       if (parser.seen_test('C')) {
-        SdBaseFile file;
-        const uint16_t size = sizeof(mc_zip);
+        MediaFile file;
         // Need to create the config size on the SD card
-        if (file.open(card.getroot(), "mc.zip", O_WRITE|O_CREAT) && file.write(pgm_read_ptr(mc_zip), size) != -1 && file.close())
-          SERIAL_ECHO_MSG("Configuration saved as 'mc.zip'");
+        MediaFile root = card.getroot();
+        if (file.open(&root, "mc.zip", O_WRITE|O_CREAT)) {
+          bool success = true;
+          for (uint16_t i; success && i < sizeof(mc_zip); ++i) {
+            const uint8_t c = pgm_read_byte(&mc_zip[i]);
+            file.write(c);
+          }
+          success = file.close() && success;
+
+          if (success) SERIAL_ECHO_MSG("Configuration saved as 'mc.zip'");
+        }
       }
     #endif
   }

--- a/Marlin/src/gcode/eeprom/M500-M504.cpp
+++ b/Marlin/src/gcode/eeprom/M500-M504.cpp
@@ -71,7 +71,7 @@ void GcodeSuite::M502() {
         MediaFile root = card.getroot();
         if (file.open(&root, "mc.zip", O_WRITE|O_CREAT)) {
           bool success = true;
-          for (uint16_t i; success && i < sizeof(mc_zip); ++i) {
+          for (uint16_t i = 0; success && i < sizeof(mc_zip); ++i) {
             const uint8_t c = pgm_read_byte(&mc_zip[i]);
             file.write(c);
           }

--- a/buildroot/share/PlatformIO/scripts/signature.py
+++ b/buildroot/share/PlatformIO/scripts/signature.py
@@ -56,13 +56,14 @@ def compute_build_signature(env):
     files_to_keep = [ 'Marlin/Configuration.h', 'Marlin/Configuration_adv.h' ]
 
     build_path = Path(env['PROJECT_BUILD_DIR'], env['PIOENV'])
+    build_path_relative = Path('.pio', 'build', env['PIOENV'])
 
     # Check if we can skip processing
     hashes = ''
     for header in files_to_keep:
         hashes += get_file_sha256sum(header)[0:10]
 
-    marlin_json = build_path / 'marlin_config.json'
+    marlin_json = build_path_relative / 'marlin_config.json'
     marlin_zip = build_path / 'mc.zip'
 
     # Read existing config file

